tests/test_flags.py

```diff
@@ -179,3 +179,55 @@ def test(self, env):
         spawn.expect(pexpect.EOF)
         spawn.close()
         assert spawn.exitstatus == 0
+
+def test_xfail_feature(tmpdir):
+    conf = tmpdir.join("config.yaml")
+    conf.write(
+"""
+targets:
+  test1:
+    features:
+      - test
+"""
+    )
+    test = tmpdir.join("test.py")
+    test.write(
+"""
+import pytest
+
+@pytest.mark.lg_xfail_feature("test")
+def test(env):
+    assert False
+"""
+    )
+
+    with pexpect.spawn(f'pytest --lg-env {conf} {test}') as spawn:
+        spawn.expect("1 xfailed")
+        spawn.expect(pexpect.EOF)
+    assert spawn.exitstatus == 0
+
+def test_no_xfail_feature(tmpdir):
+    conf = tmpdir.join("config.yaml")
+    # target without feature flags
+    conf.write(
+"""
+targets:
+  test1:
+    drivers: {}
+"""
+    )
+    test = tmpdir.join("test.py")
+    test.write(
+"""
+import pytest
+
+@pytest.mark.lg_xfail_feature("test")
+def test(env):
+    assert False
+"""
+    )
+
+    with pexpect.spawn(f'pytest --lg-env {conf} {test}') as spawn:
+        spawn.expect("1 failed")
+        spawn.expect(pexpect.EOF)
+    assert spawn.exitstatus == 1
```

Comment:
We should test the `UsageError` as well.

Reply:
Done.